File: c atau c++/NIM.Array.cpp

#include <iostream>
#include <string>
#include <vector>
#include <numeric>  
#include <algorithm> 
#include <cmath>     
#include <iomanip>   

void print_digit_array(const std::vector<int>& arr) {
    std::cout << "[";
    for (size_t i = 0; i < arr.size(); ++i) {
        std::cout << arr[i] << (i < arr.size() - 1 ? ", " : "");
    }
    std::cout << "]" << std::endl;
}

int main() {
    std::string nim_string; 
    std::cout << "========================================" << std::endl;
    std::cout << "PROGRAM ANALISIS DIGIT NIM" << std::endl;
    std::cout << "========================================" << std::endl;
    std::cout << "Masukkan NIM Anda: ";
    std::cin >> nim_string; 
    
    std::vector<int> digit_array;
    for (char c : nim_string) {
        if (std::isdigit(c)) {
            digit_array.push_back(c - '0'); 
        }
    }

    int N = digit_array.size();

    if (N == 0) {
        std::cerr << "\nError: NIM tidak mengandung digit angka yang valid." << std::endl;
        return 1;
    }
    
    std::cout << "\nNIM (String): " << nim_string << std::endl;
    std::cout << "Array Digit: ";
    print_digit_array(digit_array);
    std::cout << "Jumlah Digit: " << N << std::endl;
    std::cout << "----------------------------------------" << std::endl;
    
    int total_digit = std::accumulate(digit_array.begin(), digit_array.end(), 0);
    std::cout << "Total Seluruh Digit: " << total_digit << std::endl;
    
    int maks_digit = *std::max_element(digit_array.begin(), digit_array.end());
    std::cout << "Maksimum Digit: " << maks_digit << std::endl;

    
    int min_digit = *std::min_element(digit_array.begin(), digit_array.end());
    std::cout << "Minimum Digit: " << min_digit << std::endl;

    
    double rata_rata = (double)total_digit / N;
    std::cout << std::fixed << std::setprecision(4); 
    std::cout << "Rata-rata Digit: " << rata_rata << std::endl;

    std::cout << "----------------------------------------" << std::endl;

    std::vector<int> reversed_digit_array = digit_array;
    std::reverse(reversed_digit_array.begin(), reversed_digit_array.end());
    
    std::cout << "========================================" << std::endl;

    return 0;
}

File: Java/NIM_Array.java

import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;
import java.util.stream.Collectors;

public class NIM_Array {
    public static void printDigitArray(ArrayList<Integer> arr) {
        System.out.print("[");
        for (int i = 0; i < arr.size(); i++) {
            System.out.print(arr.get(i) + (i < arr.size() - 1 ? ", " : ""));
        }
        System.out.println("]");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String nimString;
        
        System.out.println("========================================");
        System.out.println("PROGRAM ANALISIS DIGIT NIM");
        System.out.println("========================================");
        System.out.print("Masukkan NIM Anda: ");
        nimString = scanner.nextLine(); 

        ArrayList<Integer> digitArray = new ArrayList<>();
        for (int i = 0; i < nimString.length(); i++) {
            char c = nimString.charAt(i);
            if (Character.isDigit(c)) {
                digitArray.add(c - '0'); 
            }
        }
        
        int N = digitArray.size();
        
        if (N == 0) {
            System.err.println("\nError: NIM tidak mengandung digit angka yang valid.");
            scanner.close();
            return;
        }
        
        System.out.println("\nNIM (String): " + nimString);
        System.out.print("Array Digit: ");
        printDigitArray(digitArray);
        System.out.println("Jumlah Digit: " + N);
        System.out.println("----------------------------------------");
        int totalDigit = 0;
        for (int digit : digitArray) {
            totalDigit += digit;
        }
        System.out.println("Total Seluruh Digit: " + totalDigit);
        int maksDigit = Collections.max(digitArray);
        System.out.println("Maksimum Digit: " + maksDigit);
        int minDigit = Collections.min(digitArray);
        System.out.println("Minimum Digit: " + minDigit);
        double rataRata = (double)totalDigit / N;
        System.out.printf("Rata-rata Digit: %.4f%n", rataRata); // Menggunakan printf untuk format presisi

        System.out.println("----------------------------------------");
        ArrayList<Integer> reversedDigitArray = new ArrayList<>(digitArray); 
        Collections.reverse(reversedDigitArray); // Membalik urutan secara in-place pada salinan

        System.out.print("Urutan Digit Terbalik: ");
        printDigitArray(reversedDigitArray);
        
        System.out.println("========================================");

        scanner.close();
    }
}
